lib/twitter.js

```diff
@@ -32,26 +33,42 @@ function Twitter (options) {
       headers: {
         'Accept': '*/*',
         'Connection': 'close',
-        'User-Agent': 'node-twitter/' + VERSION
+        'User-Agent': 'node-twitter/' + VERSION,
       }
     }
   }, options);
-
-  // Build a request object
-  this.request = request.defaults(
-    extend(
-      // Pass the client submitted request options
-      this.options.request_options,
-      {
-        oauth: {
-          consumer_key: this.options.consumer_key,
-          consumer_secret: this.options.consumer_secret,
-          token: this.options.access_token_key,
-          token_secret: this.options.access_token_secret
+  //Check to see if we are going to use User Authentication or Application Authetication
```

Comment:
To avoid duplication here, pull the request outside of the conditional

```js
var request_options = {
  oauth: {
  consumer_key: this.options.consumer_key,
  consumer_secret: this.options.consumer_key,
  token: this.options.access_token_key,
  token_secret: this.options.access_token_secret
}

if (this.options.bearer_token !== null) {
  request_options = {
    headers: {
      Authorization: 'Bearer ' + this.options.bearer_token
    }
}

this.request = request.defaults(
  extend(
    //Pass the client submitted request options
    this.options.request_options,
    request_options
  )
);
```